### PR TITLE
Malfunctioning AI upgrade - Action Buttons!

### PR DESCRIPTION
**Pull request by Arkatos1**
**What does this PR do:**
This PR refactors Malfunctioning AI and moves their abilities from simple verbs in the Malfunction tab to Action Buttons. Each ability has its own unique icon now. Ability descriptions were updated to be more helpful, as you can see on the example image below:

MalfButtonsExample

Some of the abilities now also play thematic sounds to the AI, for example AI Turret Upgrade module will play short construction sound to hint that turret upgrades were applied sucessfully.

Ported from /tg/station13.

Tested locally without any issue.

**Changelog:**
:cl: Arkatos
add: Traitor AI now uses action buttons for its malfunction abilities
/:cl:

### CODE REVIEW

**Comment by AzuleUtama** [comment]
Neat!

**Comment by Muhdi345** [comment]
I hope you can still use the malfunction tab, because at all times the action buttons are too damn HUGE.

I'm still sad changelings don't have the changeling tab anymore. :(

**Comment by Shadow-Quill** [comment]
Do I smell amazing?


I doo.

File: code/game/gamemodes/malfunction/Malf_Modules.dm [hunk 1]
```diff
@@ -164,342 +352,335 @@
...
+			C.toggle_cam(owner_AI, 0) //Reactivates the camera based on status. Badly named proc.
+			C.view_range = initial(C.view_range)
+			fixed_cameras++
+			uses-- //Not adjust_uses() so it doesn't automatically delete or show a message
+	to_chat(owner, "<span class='notice'>Diagnostic complete! Cameras reactivated: <b>[fixed_cameras]</b>. Reactivations remaining: <b>[uses]</b>.</span>")
+	owner.playsound_local(owner, 'sound/items/wirecutter.ogg', 50, 0)
+	adjust_uses(0, TRUE) //Checks the uses remaining
+	if(src && uses) //Not sure if not having src here would cause a runtime, so it's here to be safe
+		desc = "[initial(desc)] There are [uses] reactivations remaining."
+		owner_AI.update_action_buttons()
```
[the remark this answers is not in the document]
Comment (Arkatos1) [reply code/game/gamemodes/malfunction/Malf_Modules.dm]: Fixed.

File: code/game/gamemodes/malfunction/Malf_Modules.dm [hunk 2]
```diff
@@ -1,46 +1,241 @@
...
+						temp = AM.description
+						if(AM.one_purchase)
+							possible_modules -= AM
+						if(AM.unlock_text)
+							to_chat(A, AM.unlock_text)
+						if(AM.unlock_sound)
+							A.playsound_local(A, AM.unlock_sound, 50, 0)
+					else //Adding uses to an existing module
+						action.uses += initial(action.uses)
+						temp = "Additional use[action.uses > 1 ? "s" : ""] added to [action.name]!"
```
[the remark this answers is not in the document]
Comment (Arkatos1) [reply code/game/gamemodes/malfunction/Malf_Modules.dm]: Done.

File: code/game/gamemodes/malfunction/Malf_Modules.dm [hunk 3]
```diff
@@ -164,342 +352,335 @@
...
+/datum/action/innate/ai/blackout/Activate()
+	for(var/obj/machinery/power/apc/apc in GLOB.apcs)
+		if(prob(30 * apc.overload))
+			apc.overload_lighting()
 		else
-			to_chat(src, "<span class='warning'>Out of uses.</span>")
-
+			apc.overload++
+	to_chat(owner, "<span class='notice'>Overcurrent applied to the powernet.</span>")
+	owner.playsound_local(owner, "sparks", 50, 0)
```
[the remark this answers is not in the document]
Comment (Arkatos1) [reply code/game/gamemodes/malfunction/Malf_Modules.dm]: Done.

**Pull request merged**